### DISCUSSION TITLE
How to get BottomSheet height?

### DISCUSSION THREAD

**Discussion by vytautas-pranskunas-** [taiga-family/taiga-ui#9857] [Q&A] [answered]
I have tabs and grid inside bottom sheet. Grid is 100% height and i need to make scrollable. In order to do that i need to add height to the content block. I see that bottom sheet has ```height: calc(1197px - 14rem);```. But i cannot find a place where it sets this height. Is there a way to get this height from bottom sheet because as i understand it is dynamic.

**Comment by waterplea** [comment, marked as answer]
`SheetDialog` maximum height is entire viewport minus offset from top that you've set when you opened it. It is available as CSS variable `--tui-offset`. A rule of thumb — if CSS variable is prefixed as `--t-` — it's private API, if it's prefixed as `--tui-` — it means it's public and you can use it for your code, it's not going to change until major version where breaking changes are possible. You can then get the height using viewport size. On mobile viewport is dynamic due to address bar popping in and out, screen keyboard and other possible changes. If you need to account for that, instead of using `vh` units you can use variables listed in the last `SheetDialog` example, we update those in `tui-root` by monitoring `visualViewport` changes:
https://taiga-ui.dev/components/sheet-dialog#app-bar
